File: animationDemo/animationDemo/ViewController.m
//
//  ViewController.m
//  animationDemo
//
//  Created by cai on 2018/5/24.
//  Copyright © 2018年 cai. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()<CAAnimationDelegate>
@property (nonatomic, strong)UIImageView * imageview;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}
- (IBAction)clicke:(id)sender {
    UIImageView * imageView = [[UIImageView alloc]init];
    self.imageview = imageView;
    imageView.backgroundColor = [UIColor redColor];
    imageView.frame = CGRectMake(100, 100, 100, 100);
    imageView.transform = CGAffineTransformMakeRotation(M_PI_4);
    [self.view addSubview:imageView];
    
    CABasicAnimation *animation = [CABasicAnimation animationWithKeyPath:@"position"];
    animation.duration = 2;
    animation.repeatCount = 1;
    animation.beginTime =CACurrentMediaTime() + 1;// 1秒后执行
    animation.fromValue = [NSValue valueWithCGPoint:imageView.layer.position]; // 起始帧
    
    animation.toValue = [NSValue valueWithCGPoint:CGPointMake(imageView.layer.position.x, -100)]; // 终了帧
    
    animation.removedOnCompletion = NO;
    animation.fillMode = kCAFillModeForwards;
    
    animation.delegate = self;
    
    // 视图添加动画
    
    [imageView.layer addAnimation:animation forKey:@"move-layer"];
    
}

- (void)animationDidStop:(CAAnimation *)anim finished:(BOOL)flag{
    [self.imageview removeFromSuperview];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Broadcast/BroadcastService/BroadcastService/ViewController.swift
//
//  ViewController.swift
//  BroadcastService
//
//  Created by 李玉峰 on 2018/4/25.
//  Copyright © 2018年 cai. All rights reserved.
//

import UIKit
import ReplayKit

class ViewController: UIViewController,RPBroadcastActivityViewControllerDelegate {

    var broadcastController:RPBroadcastController? = nil
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func start(_ sender: UIButton) {
        if sender.isSelected {
            sender.isSelected = false
            guard (self.broadcastController == nil) else {
                self.broadcastController?.finishBroadcast(handler: { (error) in
                    print(error as Any)
                })
                return
            }
        }else{
            RPBroadcastActivityViewController .load { (activity, error) in
                if error == nil {
                    activity?.delegate = self as RPBroadcastActivityViewControllerDelegate
                    self .present(activity!, animated: true, completion: nil)
                }
            }
        }
    }
    
    func broadcastActivityViewController(_ broadcastActivityViewController: RPBroadcastActivityViewController, didFinishWith broadcastController: RPBroadcastController?, error: Error?) {
        broadcastActivityViewController.dismiss(animated: true, completion: nil)
        print(broadcastController as Any,error as Any)
        self.broadcastController = broadcastController
        broadcastController?.startBroadcast(handler: { (error) in
            print(error as Any)
        })
    }
    
    
}

